src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = Object.assign({}, state);
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ......... } = action;
    
    switch (type) {
```